impl/src/main/java/org/jboss/weld/injection/producer/NoopLifecycleCallbackInvoker.java

```diff
@@ -55,6 +55,12 @@ public boolean hasPreDestroyMethods() {
 
     @Override
     public boolean hasPostConstructMethods() {
-        return true; // we cannot say no for sure
+        return hasPostConstructCallback();
```

Comment:
This method should return `true` as before - we still cannot say for sure... Otherwise the new `hasPostConstructCallback()` would be useless and we could just override this one.